fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -596,25 +597,36 @@ public void deleteFile(@Nonnull String name) throws IOException {
             return;
         }
         try {
-            boolean deleted = Objects.requireNonNull(context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_DELETE_FILE, getFileReferenceCacheAsync().thenApply(cache -> {
-                FDBLuceneFileReference value = cache.get(name);
-                if (value == null) {
-                    return false;
-                }
-                context.ensureActive().clear(metaSubspace.pack(name));
-                context.ensureActive().clear(dataSubspace.subspace(Tuple.from(value.getId())).range());
-                cache.remove(name);
-                return true;
-            })));
+            boolean deleted = Objects.requireNonNull(context.asyncToSync(LuceneEvents.Waits.WAIT_LUCENE_DELETE_FILE, getFileReferenceCacheAsync()
+                    .thenApply(cache -> deleteFileInternal(cache, name))));
 
             if (!deleted) {
                 throw new NoSuchFileException(name);
             }
+
+            if (isCompoundFile(name)) {
+                Map<String, FDBLuceneFileReference> cache = fileReferenceCache.get();
+                String primaryKeyName = name.substring(0, name.length() - DATA_EXTENSION.length()) + "pky";
```

Comment:
Would it be better to use some of the methods from `IndexFileNames`?

Reply:
I'm not sure that ends up being more clear / stable. It's sort of modeled after the existing `convertToDataFile`.